packages/alert-banner/index.tsx

```diff
@@ -0,0 +1,98 @@
+import { useEffect, useState } from 'react'
+import cookie from 'js-cookie'
+import slugify from 'slugify'
+import classNames from 'classnames'
+import VisuallyHidden from '@reach/visually-hidden'
+import useProductMeta from '@hashicorp/platform-product-meta'
+import InlineSvg from '@hashicorp/react-inline-svg'
+import CloseIcon from './img/close-icon.svg?include'
+import fragment from './fragment.graphql'
+import s from './style.module.css'
+import analytics from './analytics'
+import { HashiCorpProduct } from '../../types'
+
+interface AlertBannerProps {
+  tag: string
+  text: string
+  url: string
+  expirationDate?: string
+  hideOnMobile?: boolean
+  linkText?: string
+  name?: string
+  product?: HashiCorpProduct
+}
+
+/**
+ * AlertBanner is a component used widely across HashiCorp websites to
+ * draw attention to an important release, event, etc. It appears in
+ * bold colors at the top of the website, and can be dismissed.
+ */
+function AlertBanner({
+  expirationDate,
+  hideOnMobile,
+  linkText,
+  name,
+  product = 'vagrant', // override useProductMeta's default "hashicorp", as it would result in a dull gray background
+  tag,
+  text,
+  url,
+}: AlertBannerProps): React.ReactElement {
+  const dismissalCookieId = `banner_${name || slugify(text, { lower: true })}`
+  const [isShown, setIsShown] = useState(true)
+  const { themeClass } = useProductMeta(product)
+
+  /**
+   * On mount, hide the banner if it is expired or
+   * if a cookie indicates it was previously closed
+   */
+  useEffect(() => {
+    const hasBeenDismissed = cookie.get(dismissalCookieId)
+    const hasExpired = expirationDate && Date.now() > Date.parse(expirationDate)
+    const shouldBeShown = !hasBeenDismissed && !hasExpired
+    if (isShown !== shouldBeShown) setIsShown(shouldBeShown)
```

Comment:
I think _technically_ `isShown` should be in the dependency array here. We can use a functional updater to avoid this, as well as drop the if and the need to reference `isShown` directly:

```tsx
setIsShown(current => current !== shouldBeShown ? shouldBeShown : current)
```

React will skip updates if the new value is the same:

> If your update function returns the exact same value as the current state, the subsequent rerender will be skipped completely.

https://reactjs.org/docs/hooks-reference.html#usestate

Reply:
✅ Updated